src/System.Linq/tests/SelectTests.cs

```diff
@@ -1168,5 +1168,65 @@ public static IEnumerable<object[]> MoveNextAfterDisposeData()
             yield return new object[] { new int[1] };
             yield return new object[] { Enumerable.Range(1, 30) };
         }
+
+        [Theory]
+        [MemberData(nameof(RunSelectorDuringCountData))]
+        public void RunSelectorDuringCount(IEnumerable<int> source)
+        {
+            int timesRun = 0;
+            var selected = source.Select(i => timesRun++);
+            selected.Count();
+
+            Assert.Equal(source.Count(), timesRun);
+        }
+
+        // [Theory]
```

Comment:
Disabled currently because the first assert is giving inconsistent results. See comment above